File: src-tauri/src/commands/external/seventv_events.rs
use crate::services::external::seventv_events::{SevenTvEvents, SevenTvOp};
use serde::Deserialize;

#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SeventvEmoteSetParams {
    pub emote_set_id: String,
}

#[tauri::command]
pub fn seventv_subscribe_emote_set(
    state: tauri::State<SevenTvEvents>,
    params: SeventvEmoteSetParams,
) {
    let _ = state.0.send(SevenTvOp::Subscribe(params.emote_set_id));
}

#[tauri::command]
pub fn seventv_unsubscribe_emote_set(
    state: tauri::State<SevenTvEvents>,
    params: SeventvEmoteSetParams,
) {
    let _ = state.0.send(SevenTvOp::Unsubscribe(params.emote_set_id));
}

File: src/lib/api/external/seventv_events.ts
import { invokeCommand, type InvokeOptions } from "../utils";

export type SeventvEmoteSetParams = {
  emoteSetId: string;
};

export function seventvSubscribeEmoteSet(
  params: SeventvEmoteSetParams,
  options?: InvokeOptions,
): Promise<void> {
  return invokeCommand("seventv_subscribe_emote_set", params, options);
}

export function seventvUnsubscribeEmoteSet(
  params: SeventvEmoteSetParams,
  options?: InvokeOptions,
): Promise<void> {
  return invokeCommand("seventv_unsubscribe_emote_set", params, options);
}
